app/views/dashboard/index.html.erb

```diff
@@ -1,4 +1,11 @@
 <h1>Dashboard</h1>
-<%= link_to "Log Out", logout_path, method: :post %>
+<% if @user.username.nil? || @user.zip.nil? %>
+	<h3>Action Needed!</h3>
```

Comment:
One option for tackling the edge case of user going into "/dashboard", "/profile", without having completed the registration requirement.